languages/elixir/exercises/concept/exceptions/lib/rpn_calculator.ex

```diff
@@ -0,0 +1,15 @@
+defmodule RPNCalculator.Exception do
+  defmodule DivisionByZeroError do
+    raise "Please implement the DivisionByZeroError exception"
+  end
+
+  defmodule StackUnderflowError do
+    raise "Please implement the StackUnderflowError exception"
```

Comment:
This error (same as the one on line 7) is raised compile-time. That means that it won't be possible for people to do step-by-step development where they first attempt to implement `DivisionByZeroError`, run the tests, fix their implementation, run the tests again, and only then start working on `StackUnderflowError`.

To allow for such a workflow, we could change those two `raises` to just comments - the tests will fail regardless. However, the tests itself would also need to be changed to use `struct(ModuleName, %{})` instead `%ModuleName{}` because the latter also raises a compile-time error if `ModuleName` is not a struct.

Reply:
There may be a way to accomplish this similar to the structs exercise where I quote the code. I will have to experiment

Reply:
So my solution is this:
in the test_helper (which is run first by `mix test`) checks to see if the error modules have been defined, and if their __struct__/0 function has been created by `defstruct`.  If they havent this changes what tests are run in the test file.

Pros: have test cases and errors for each state without compiler errors occuring before the student has had a chance to write the module.

Cons: the test suite file is less clear, is a little non-standard.

Thoughts?